SDK v2 migration guide.md

```diff
@@ -0,0 +1,399 @@
+| ModuleClient#subscribeToMethod(DeviceMethodCallback , Object, IotHubEventCallback, Object); | ModuleClient#subscribeToMethodsAsync(MethodCallback, Object, IotHubEventCallback, Object);  |
+| ModuleClient#startTwin(IotHubEventCallback, Object, TwinPropertyCallback, Object); | ModuleClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| ModuleClient#startTwin(IotHubEventCallback, Object, TwinPropertiesCallback, Object); | ModuleClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| ModuleClient#startTwin(IotHubEventCallback, Object, PropertyCallBack<Type1, Type2>, Object); | ModuleClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| ModuleClient#subscribeToDesiredProperties(Map<Property, Pair<PropertyCallBack<String, Object>, Object>>); | ModuleClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| ModuleClient#subscribeToTwinDesiredProperties(Map<Property, Pair<TwinPropertyCallBack, Object>>(); | ModuleClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| ModuleClient#sendReportedProperties(Set<Property>); | ModuleClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesCallback, Object);  |
+| ModuleClient#sendReportedProperties(Set<Property>, int); | ModuleClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesCallback, Object);  |
+| ModuleClient#sendReportedProperties(ReportedPropertiesParameters); | ModuleClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesUpdateCorrelatingMessageCallback, Object);  |
+| ModuleClient#sendReportedProperties(Set<Property>, Integer, CorrelatingMessageCallback, Object, IotHubEventCallback, Object); | ModuleClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesUpdateCorrelatingMessageCallback, Object);  |
+| ModuleClient#setOption(String, Object); | no equivalent method***  |
+| ModuleClient#setProxySettings(ProxySettings); | no equivalent method****  |
+
+*** The options that were previously set in this method are now set at ModuleClient constructor time in the optional ClientOptions parameter.
+
+**** Proxy settings are now set at ModuleClient constructor time in the optional ClientOptions parameter,
+
+#### MultiplexingClient
+
+No API surface changes have been made to the MultiplexingClient class
+
+#### TransportClient
+
+This client has been removed in v2. It is replaced by the MultiplexingClient. See this repo's multiplexing client sample for more information.
+
+#### Other notable breaking changes
+
+- Trust certificates are read from the physical device's trusted root certification authorities certificate store rather than from source.
```

Comment:
I think seeing the APIs side-by-side with this description might be useful. Otherwise, people will need to look through the client to figure out which API we are talking about here.

What do you thik?

Reply:
This is a tricky one since there is no API for loading certificates. We do that for the user when they construct their client and open a connection to Hub/DPS. I can expand a bit on this in the notes here and clarify that, for most users, no action is required for them here. After all, the IoT hub public certs are typically already installed on every device's cert store.